core/src/main/java/org/elasticsearch/common/logging/LogConfigurator.java

```diff
@@ -79,9 +103,25 @@ public static void configureWithoutConfig(final Settings settings) {
      */
     public static void configure(final Environment environment) throws IOException, UserException {
         Objects.requireNonNull(environment);
+        try {
+            checkErrorListener();
+        } finally {
+            StatusLogger.getLogger().removeListener(LISTENER);
+        }
         configure(environment.settings(), environment.configFile(), environment.logsFile());
     }
 
+    private static void checkErrorListener() {
+        assert errorListenerIsRegistered() : "expected error listener to be registered";
+        if (error.get()) {
+            throw new IllegalStateException("status logger logged an error before logging was configured");
```

Comment:
Maybe change the message to something like "used logging before it was configured"?